frontend/src/components/Candidate-Decider/CandidateDecider.tsx

```diff
@@ -136,29 +172,39 @@ const CandidateDecider: React.FC<CandidateDeciderProps> = ({ uuid }) => {
               text: candidate.id
             }))}
             onChange={(_, data) => {
-              setCurrentCandidate(data.value as number);
-              populateReviewForCandidate(data.value as number);
+              handleCandidateChange(data.value as number);
             }}
           />
           <span className={styles.ofNum}>of {instance.candidates.length}</span>
           <Button.Group className={styles.previousNextButtonContainer}>
-            <Button basic color="blue" disabled={currentCandidate === 0} onClick={previous}>
+            <Button
+              basic
+              color="blue"
+              disabled={currentCandidate === 0}
+              onClick={() => {
+                if (currentCandidate === 0) return;
+                setNextCandidate(currentCandidate - 1);
```

Comment:
How about let's move `setNextCandidate` into `handleCandidateChange`? And move the bound checks there as well (==0 and == instance.candidates.length - 1)?